tests/fixture/bootstrapmonitor/wait.go

```diff
@@ -0,0 +1,194 @@
+	}
+	log.Info("Retrieved bootstrap test config", zap.Reflect("testConfig", testConfig))
+
+	// Avoid checking node health before it reports initial ready
+	log.Info("Waiting for pod readiness")
+	if err := WaitForPodCondition(ctx, clientset, namespace, podName, corev1.PodReady); err != nil {
+		return fmt.Errorf("failed to wait for pod condition: %w", err)
+	}
+
+	log.Info("Waiting for node to report healthy")
+	if err := wait.PollImmediateInfinite(healthCheckInterval, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), contextDuration)
+		defer cancel()
+
+		// Define common fields for logging
+		diskUsage := getDiskUsage(log, dataDir)
+		commonFields := []zap.Field{
+			zap.String("diskUsage", diskUsage),
+			zap.Duration("duration", time.Since(testDetails.StartTime)),
+		}
+
+		// Check whether the node is reporting healthy which indicates that bootstrap is complete
+		if healthy, err := tmpnet.CheckNodeHealth(ctx, nodeURL); err != nil {
+			log.Error("failed to check node health", zap.Error(err))
+			return false, nil
+		} else {
+			if !healthy.Healthy {
+				log.Info("Node reported unhealthy", commonFields...)
+				return false, nil
+			}
+
+			log.Info("Node reported healthy")
+		}
+
+		commonFields = append(commonFields, zap.Reflect("testConfig", testConfig))
+		log.Info("Bootstrap completed successfully", commonFields...)
+
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to wait for node to report healthy: %w", err)
+	}
+
+	log.Info("Waiting for new image to test")
+	if err := wait.PollImmediateInfinite(imageCheckInterval, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), contextDuration)
+		defer cancel()
+
+		log.Info("Starting pod to get the image id for the `latest` tag")
+		latestImageDetails, err := getLatestImageDetails(ctx, log, clientset, namespace, testConfig.Image, nodeContainerName)
+		if err != nil {
+			log.Error("failed to get latest image id", zap.Error(err))
+			return false, nil
+		}
+
+		if latestImageDetails.Image == testConfig.Image {
+			log.Info(ImageUnchanged)
+			return false, nil
+		}
+
+		log.Info("Found updated image",
+			zap.String("image", latestImageDetails.Image),
+			zap.Reflect("versions", latestImageDetails.Versions),
+		)
+
+		log.Info("Updating StatefulSet to trigger a new test")
+		if err := setImageDetails(ctx, log, clientset, namespace, podName, latestImageDetails); err != nil {
+			log.Error("failed to set container image", zap.Error(err))
+			return false, nil
+		}
+
+		// Statefulset will restart the pod with the new image
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to wait for new image to test: %w", err)
+	}
+
+	// Avoid exiting immediately to avoid container restart before the pod is recreated with the new image
+	time.Sleep(5 * time.Minute)
+	return nil
+}
+
+// Determines the current disk usage for the specified directory
+func getDiskUsage(log logging.Logger, dir string) string {
```

Comment:
I'm sure that it would work.. but it's also very error prone.
I'd opt into something like :
```golang
func DirSize(path string) (int64, error) {
    var size int64
    err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
        if err != nil {
            return err
        }
        if !info.IsDir() {
            size += info.Size()
        }
        return err
    })
    return size, err
}
```

Reply:
I generally dislike writing linux/darwin specific code when there are equivalent solutions in the GO ecosystem.

I'm going to resolve this one, since I don't want this to block you.